### PR TITLE
gluu-openid-connect plugin - initial version, #283

### PR DESCRIPTION
**Pull request by altexy**
This PR is just for preliminary code review, don't merge without working tests

### CODE REVIEW

File: kong/plugins/gluu-openid-connect/handler.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
+local BasePlugin = require "kong.plugins.base_plugin"
+local access = require "kong.plugins.gluu-openid-connect.access"
+
+local handler = BasePlugin:extend()
+handler.PRIORITY = 999
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-openid-connect/handler.lua]: ok

File: kong/plugins/gluu-openid-connect/schema.lua [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        op_url = { required = true, type = "url" },
+        authorization_redirect_path = { required = true, type = "url" }, -- TODO any default?!
+        logout_path = { required = false, type = "string" },
+        logout_pass_upstream = { required = false, type = "boolean", default = false },
+        post_logout_redirect_uri = { required = false, type = "url" }, --TODO must be registered as well as authorization_redirect_uri
+        requested_scopes = {required = true, type = "string"},
+        required_acrs = {required = false, type = "string"}, -- UI should show associated levels
+        -- required_auth_level = { required = false, type = "number" }, -- on hold
+        max_id_token_age = { required = true, type = "timestamp"},
+        max_id_token_auth_age = { required = true, type = "timestamp"},
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-openid-connect/schema.lua]: timestamp is an integer number, IMO both acceptable

File: kong/plugins/gluu-openid-connect/schema.lua [hunk 3]
```diff
@@ -0,0 +1,21 @@
+return {
+    no_consumer = true,
+    fields = {
+        -- hide_credentials = { type = "boolean", default = false }, TODO not sure
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-openid-connect/schema.lua]: ok

File: kong/plugins/gluu-openid-connect/schema.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (ldeveloperl1985) [reply kong/plugins/gluu-openid-connect/schema.lua]: Okay

File: kong/plugins/gluu-openid-connect/schema.lua [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        oxd_id = { required = true, type = "string" },
+        oxd_url = { required = true, type = "url" },
+        client_id = { required = true, type = "string" },
+        client_secret = { required = true, type = "string" },
+        op_url = { required = true, type = "url" },
+        authorization_redirect_path = { required = true, type = "url" }, -- TODO any default?!
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-openid-connect/schema.lua]: I have removed the TODO - IMO it must be just configured

File: kong/plugins/gluu-openid-connect/access.lua [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+    session_data.access_token = json.access_token
+    session_data.access_token_expiration = ngx.time() + access_token_expires_in(conf, json.expires_in)
+    session_data.refresh_token = json.refresh_token
+
+    local ptoken = kong_auth_pep_common.get_protection_token(nil, conf)
+
+    local response, err = oxd.get_user_info(conf.oxd_url,
+        {
+            oxd_id = conf.oxd_id,
+            access_token = access_token,
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-openid-connect/access.lua]: thanks

File: kong/plugins/gluu-openid-connect/schema.lua (hunk 4 above)
[the remark this answers is not in the document]
Comment (ldeveloperl1985) [reply kong/plugins/gluu-openid-connect/schema.lua]: Okay

File: t/specs/gluu-openid-connect/test_spec.lua [hunk 6]
```diff
@@ -185,5 +187,12 @@ test("basic", function()
...
 
+    print"logout and check the cookie"
+    local res, err = sh_ex([[curl -i --fail -sS -X GET --url http://localhost:]],
+        ctx.kong_proxy_port, [[/logout_path --header 'Host: backend.com' -c ]], cookie_tmp_filename,
+        [[ -b ]], cookie_tmp_filename)
+    assert(res:find("302", 1, true))
+    assert(res:find("session=;", 1)) -- no cookie is available
+
     ctx.print_logs = false -- comment it out if want to see logs
 end)
```
Comment (altexy) [inline t/specs/gluu-openid-connect/test_spec.lua]: Please, add newline here, all others looks good

Comment (ldeveloperl1985) [reply t/specs/gluu-openid-connect/test_spec.lua]: Okay

**Pull request merged**